x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisFailureIT.java

```diff
@@ -160,16 +160,28 @@ public void testFailsOnChecksumMismatch() {
         blobStore.setDisruption(new Disruption() {
             @Override
             public byte[] onRead(byte[] actualContents, long position, long length) {
-                final byte[] disruptedContents = actualContents == null ? null : Arrays.copyOf(actualContents, actualContents.length);
-                if (actualContents != null && countDown.countDown()) {
-                    // CRC32 should always detect a single bit flip
-                    disruptedContents[Math.toIntExact(position + randomLongBetween(0, length - 1))] ^= (byte) (1 << between(0, 7));
+                final byte[] disruptedContents;
+                final boolean isCountedDown = countDown.countDown();
+                if (isCountedDown) {
+                    if (actualContents != null) {
+                        disruptedContents = Arrays.copyOf(actualContents, actualContents.length);
+                        // CRC32 should always detect a single bit flip
+                        disruptedContents[Math.toIntExact(position + randomLongBetween(0, length - 1))] ^= (byte) (1 << between(0, 7));
+                    } else {
+                        disruptedContents = randomByteArrayOfLength(1);
+                    }
+                } else {
+                    disruptedContents = actualContents;
                 }
                 return disruptedContents;
             }
         });
 
-        expectThrows(RepositoryVerificationException.class, () -> analyseRepository(request));
+        expectThrows(
+            RepositoryVerificationException.class,
+            "countDown has reached to zero: " + countDown.isCountedDown(),
```

Comment:
I think it is worth moving to a dedicated assertion prior this one.

Reply:
I thought about that. But it is not possible to assert `countDown` without excuting the analysis. But once the analysis is executed we need to check the error it throws first. So there is an order issue.